generated/sql/microsoft.sql/2017-10-01-preview/types.md

```diff
@@ -135,23 +135,23 @@
 ## ManagedInstanceEncryptionProtectorProperties
 ### Properties
 * **serverKeyName**: string: The name of the managed instance key.
-* **serverKeyType**: 'AzureKeyVault' | 'ServiceManaged' (Required, WriteOnly): The key type like 'ServiceManaged', 'AzureKeyVault'.
```

Comment:
There have been no changes to Microsoft.Sql@2017-10-01-preview [since April 2021](https://github.com/Azure/azure-rest-api-specs/commits/main/specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview), so the removal of the `WriteOnly` flag must be a consequence of some change in the way models are loaded in Autorest.

The `WriteOnly` flag was correct (this property [has an `x-ms-mutability` extension with a value of `["create"]`](https://github.com/Azure/azure-rest-api-specs/blob/2f7b225e23442d0ade36bf15d4ac15730b683c9c/specification/sql/resource-manager/Microsoft.Sql/preview/2017-10-01-preview/ManagedInstanceKeys.json#L264)), so it's not clear what's causing this change. There have been no changes to how this repo handles x-ms-mutability extensions since https://github.com/Azure/bicep-types-az/pull/436

Reply:
Tracking as #836